eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java

```diff
@@ -126,71 +140,91 @@ protected void leaderStart() {
     client.setBinlogPosition(bfo.getOffset());
 
     client.setEventDeserializer(getEventDeserializer());
-    client.registerEventListener(event -> {
-      switch (event.getHeader().getEventType()) {
-        case TABLE_MAP: {
-          TableMapEventData tableMapEvent = event.getData();
-
-          if (cdcMonitoringDao.isMonitoringTableChange(tableMapEvent.getDatabase(), tableMapEvent.getTable())) {
-            cdcMonitoringTableId = Optional.of(tableMapEvent.getTableId());
-            tableMapEventByTableId.put(tableMapEvent.getTableId(), tableMapEvent);
-            break;
-          }
+    client.registerEventListener(event -> listenBinlogEventsHandleErrors(event, binlogFileOffset));
 
-          cdcMonitoringTableId = cdcMonitoringTableId.filter(id -> !id.equals(tableMapEvent.getTableId()));
+    connectWithRetriesOnFail();
 
-          SchemaAndTable schemaAndTable = new SchemaAndTable(tableMapEvent.getDatabase(), tableMapEvent.getTable());
+    try {
+      stopCountDownLatch.await();
+    } catch (InterruptedException e) {
+      handleProcessingFailException(e);
+    }
+  }
 
-          boolean shouldHandleTable = binlogEntryHandlers
-                  .stream()
-                  .map(BinlogEntryHandler::getSchemaAndTable)
-                  .anyMatch(schemaAndTable::equals);
+  private void listenBinlogEventsHandleErrors(Event event, Optional<BinlogFileOffset> binlogFileOffset) {
+    try {
+      listenBinlogEvents(event, binlogFileOffset);
+    } catch (Exception e) {
+      handleRestart(e);
+    }
+  }
 
-          if (shouldHandleTable) {
-            tableMapEventByTableId.put(tableMapEvent.getTableId(), tableMapEvent);
-          } else {
-            tableMapEventByTableId.remove(tableMapEvent.getTableId());
-          }
+  private void handleRestart(Exception e) {
+    logger.error(e.getMessage(), e.getMessage());
+    publishingException = Optional.of(e);
+    leaderSelector.stop();
+    callbackOnStop = Optional.of(() -> {
+      callbackOnStop = Optional.empty();
+      start();
+    });
+  }
 
-          dbLogMetrics.onBinlogEntryProcessed();
+  private void listenBinlogEvents(Event event, Optional<BinlogFileOffset> binlogFileOffset) {
```

Comment:
This method is  badly named. `handleBinlogEvent` would be better.

Reply:
ok